File: src/main/java/zwylair/randomitembattle/utils/PosCalculator.java
package zwylair.randomitembattle.utils;

import net.minecraft.util.math.Vec3d;

public class PosCalculator {
    public static int posToCoord(double number) {
        if (number < 0) { return (int) number - 1; } else { return (int) number; }
    }

    public static Vec3d roundPosition(Vec3d position) {
        double x = (int) position.getX();
        if (position.getX() < 0) { x -= 0.5; } else { x += 0.5; }
        double z = (int) position.getZ();
        if (position.getZ() < 0) { z -= 0.5; } else { z += 0.5; }

        return new Vec3d(x, position.getY(), z);
    }
}

File: src/main/kotlin/zwylair/randomitembattle/utils/PosCalculator.kt
package zwylair.randomitembattle.utils

import net.minecraft.util.math.Vec3d

class PosCalculator {
    companion object {
        fun posToCoord(number: Double): Int {
            return if (number < 0) (number - 1).toInt() else number.toInt()
        }

        fun roundPosition(position: Vec3d): Vec3d {
            var x = position.x.toInt().toDouble()
            x += if (position.x < 0) -0.5 else 0.5
            var z = position.z.toInt().toDouble()
            z += if (position.z < 0) -0.5 else 0.5
            return Vec3d(x, position.y, z)
        }
    }
}
